client/app/components/composites/SideWinder/SideWinderTransition.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable react/no-set-state  */
+
+import { Component, PropTypes } from 'react';
+import { div } from 'r-dom';
+
+import css from './SideWinder.css';
+
+// ReactTransitionGroup is a low level animation API that handles
+// component mounting, unmounting and leaving/entering between
+// animation states. It works by calling certain lifecycle methods on
+// its child component that can then asynchronously handle the
+// transitions to other states.
+//
+// This component implements the required lifecycle methods to add
+// state classes to its container, allowing for CSS styles to attach
+// to transitions properly.
+//
+// The component also attaches state classes to document.body for
+// styling other components outside the current render tree position.
+//
+// See: https://facebook.github.io/react/docs/animation.html#reacttransitiongroup
+class SideWinderTransition extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: true,
+      entering: false,
+      leaving: false,
+    };
+  }
+  componentWillUnmount() {
+    document.body.classList.remove(css.transitionVisible);
+  }
+  componentWillAppear(callback) {
+    callback();
+  }
+  componentWillEnter(callback) {
+    this.setState({ entering: true, open: true });
+    window.setTimeout(callback, this.props.enterTimeout);
```

Comment:
As discussed, React life cycle could cause problems with uncleared timeouts (at least, I had lots of problems with FlashMessages related to these.)